### PR TITLE
FWF-3834:[Feature]-Import Functionality in FormEdit, Subflow create a…

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
JIRA: 
Issue Type: BUG/ FEATURE

# Changes
Import in:  
-->Form Create/Edit page
-->Sub flow Listing page, Edit page
-->Decision Table Listing page, Edit page.

# Screenshots (if applicable)
image
image
image
image
image
image

### CODE REVIEW

File: forms-flow-web/src/components/Modeler/SubFlowTable.js [hunk 1]
```diff
@@ -38,34 +40,47 @@ const SubFlow = React.memo(() => {
...
   };
-  
+
+
   const handleImportClick = () => {
-    console.log("Import clicked");
+    setShowBuildModal(false);
+    setImportSubflow(true);
+    setShowBuildModal(false);
+    setImportSubflow(true);
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modeler/SubFlowTable.js:59]: duplicate lines

File: forms-flow-web/src/components/Modeler/SubFlowTable.js [hunk 2]
```diff
@@ -99,27 +114,19 @@ const SubFlow = React.memo(() => {
...
+    { text: "5", value: 5 },
+    { text: "25", value: 25 },
+    { text: "50", value: 50 },
+    { text: "100", value: 100 },
+    { text: "All", value: totalCount },
+    { text: "5", value: 5 },
+    { text: "25", value: 25 },
+    { text: "50", value: 50 },
+    { text: "100", value: 100 },
+    { text: "All", value: totalCount },
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modeler/SubFlowTable.js:126]: duplciate lines

File: forms-flow-web/src/components/Modeler/SubFlowTable.js [hunk 3]
```diff
@@ -135,12 +142,14 @@ const SubFlow = React.memo(() => {
...
+
+
   const gotoEdit = (data) => {
     if (MULTITENANCY_ENABLED) {
       dispatch(setIsPublicDiagram(!!data.tenantId));
     }
-    dispatch(push(`${redirectUrl}subflow/edit/${data.processKey}`)
-    );
+    dispatch(push(`${redirectUrl}subflow/edit/${data.processKey}`));
+    dispatch(push(`${redirectUrl}subflow/edit/${data.processKey}`));
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modeler/SubFlowTable.js:152]: duplicate lines

File: forms-flow-web/src/services/FileService.js [hunk 4]
```diff
@@ -33,27 +33,45 @@ const uploadFile = (evt, callback) => {
...
+          } else {
+            console.error("No 'forms' array found in the file.");
+            callback(null); // Pass null if the 'forms' array is missing
+          }
+        } else if (['bpmn', 'dmn'].includes(fileExtension)) {
+          // Handle XML file parsing
+          const parser = new DOMParser();
+          const xmlDoc = parser.parseFromString(e.target.result, "application/xml");
+
+          if (xmlDoc.getElementsByTagName("parsererror").length > 0) {
```
Comment (arun-s-aot) [inline forms-flow-web/src/services/FileService.js:63]: do a null check . if xmlDoc.getElementsByTagName("parsererror") is not present, then this line will break

File: forms-flow-web/src/components/Modals/ImportSubFlow.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    return new Promise((resolve, reject) => {
+      FileService.extractFileDetails(fileContent, (result) => {
+        result ? resolve(result) : reject("No valid XML found in the file.");
+      });
+    });
+  };
+
+  return (
+    <ImportModal
+      importModal={showModal}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/ImportSubFlow.js:95]: can u change importModal to showModal in MF also

File: forms-flow-web/src/components/Modals/ImportDecisionTable.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        return new Promise((resolve, reject) => {
+            FileService.extractFileDetails(fileContent, (result) => {
+                result ? resolve(result) : reject("No valid XML found in the file.");
+            });
+        });
+    };
+
+    return (
+        <ImportModal
+            importModal={showModal}
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modals/ImportDecisionTable.js:88]: can u change importModal to showModal in MF also

File: forms-flow-web/src/components/Modeler/DecisionTable.js [hunk 7]
```diff
@@ -134,110 +142,126 @@ const contents = [
...
+        show={showBuildModal}
+        onClose={handleBuildModal}
+        title={t("New DMN")}
+        contents={contents}
+      />
+      {importDecisionTable && (
+        <ImportProcess
+          showModal={importDecisionTable}
+          closeImport={closeDmnImport}
+          fileType=".bpmn"
```
Comment (arun-s-aot) [inline forms-flow-web/src/components/Modeler/DecisionTable.js:259]: pls change to .dmn @Josephalexantony-aot

**Review by arun-s-aot — APPROVED** [review]

**Review by shuhaib-aot — APPROVED** [review]

**Pull request merged**